sdp.js

```diff
@@ -551,6 +551,39 @@ SDPUtils.parseMsid = function(mediaSection) {
   }
 };
 
+// SCTP Datachannels
+// parses either the older sctpmap format, or the newer sctp-port format
+SDPUtils.parseSctpMap = function(mediaSection) {
+  var sctpMapLines = SDPUtils.matchPrefix(mediaSection, 'a=sctpmap:');
+  if (sctpMapLines.length > 0) {
+    var parts = SDPUtils.matchPrefix(mediaSection, 'a=sctpmap:')[0]
+      .substr(10)
+      .split(' ');
+    return {
+      number: parts[0],
+      protocol: parts[1],
+      streams: parts[2]
+    };
+  }
+  var sctpPort = SDPUtils.matchPrefix(mediaSection, 'a=sctp-port:');
+  return {
+    number: sctpPort[0].substr(12),
+    protocol: 'webrtc-datachannel',
+    streams: '1024'
+  };
+};
+
+// SCTP Datachannels
+// output only the sctp-port version now that all browsers support
+// receiving in this format
+SDPUtils.writeSctpDescription = function(media, sctp) {
+  return [
+    'm=' + media.kind + ' 9 ' + media.protocol + ' ' + sctp.protocol + '\r\n',
```

Comment:
`m=application 9 DTLS/SCTP 5000`
-- chrome gives me this so sctp.protocol seems wrong and should be sctp.number?

Reply:
Oh, lovely. Firefox gives me `m=application 9 UDP/DTLS/SCTP webrtc-datachannel`

Reply:
urgh. When in doubt: @lgrahl, please advise

Reply:
i think the chrome format is an old one here...

Reply:
Yeah, that's an old version line. This PR will only emit the new format, but will ingest the old one.

Reply:
For maximum fun, `TCP/DTLS/SCTP` is also valid.

Reply:
github is clearly missing the :trollface: emoji reaction :-)